hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -414,13 +413,21 @@ update Environment{party, signingKey, otherParties} ledger st ev = case (st, ev)
                    ]
             )
   --
-  (_, OnChainEvent (Observation OnContestTx{snapshotNumber})) ->
-    -- TODO: Is there more to handle contestation?
-    sameState [ClientEffect HeadIsContested{snapshotNumber}]
-  (ClosedState{utxos}, ShouldPostFanout) ->
-    sameState [OnChainEffect (FanoutTx utxos)]
-  (ClosedState{utxos}, OnChainEvent (Observation OnFanoutTx{})) ->
-    nextState ReadyState [ClientEffect $ HeadIsFinalized utxos]
+  (ClosedState{confirmedSnapshot}, OnChainEvent (Observation OnContestTx{snapshotNumber}))
```

Comment:
Do we _assume_ that the snapshot's multisig was verified somewhere outside of the head-logic (e.g. when observing the contest event)? If so, I find it a little _dodgy_ as I would rather keep the head logic free of assumptions. Thus here, we should also verify that the contest observation is legit before taking action for it. Same for close I believe.

Reply:
It seems we cannot verify anything here, the only information we have is the `snapshotNumber`

Reply:
You're right, we should probably retrieve the hash + signature from the actual transaction and pass it up.